CONTRIBUTING_A_NEW_QUEST.md

```diff
@@ -0,0 +1,390 @@
+
+# Adjust the copy
+
+After the quest is copied it is necessary to adjust it a bit.
+
+Change its class name and the file name to the new one.
+
+In copied code change package info (things like `package de.westnordost.streetcomplete.quests.defibrillator` at the top) to match the new folder containing the quest.
+
+# Add the quest to the list of active ones
+
+Adjust [QuestModule.kt](app/src/main/java/de/westnordost/streetcomplete/quests/QuestModule.kt) file. It contains a big list of active quests, ordered by priority. Read [what governs their priority](https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/QuestModule.kt#L138-L162) but do not worry too much, it can be tweaked later.
+
+Add your quest to the list so that it will be loaded by the app.
+
+## Quest anatomy
+See for example [simple yes/no quest asking whether AED is indoor or outdoor](app/src/main/java/de/westnordost/streetcomplete/quests/defibrillator/AddIsDefibrillatorIndoor.kt).
+
+
+TODO: maybe linking [app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquests/OsmElementQuestType.kt](app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquests/OsmElementQuestType.kt) would be better? And extend documentation there?
+
+### Element selection
+elementFilter property defines nodes, ways and relations which will be selected for a given quest. It is an element selection used by OsmFilterQuestType.
+
+```
+"""
+        nodes with
+         emergency = defibrillator
+         and access !~ private|no
+         and !indoor
+"""
+``` 
+
+This query will be limited to nodes (`nodes with`), which fulfill some requirements.
+
+- `emergency = defibrillator` tag must be present
+- `access` tag must not have values `private` or `no` to skip ones where mapper will be unable to survey (`!~ private|no` - to be more specific optionA|optionB is treated like `^optionA|optionB$` regexp)
+- `indoor` key must not be present at all, to show only ones where this tag is still missing
+
+It is specified as a string, in syntax specific to StreetComplete. You can look around some quests to see how it works. If you are trying to implement a new quest and you got stuck here, [open a new issue](https://github.com/streetcomplete/StreetComplete/issues) to request more thorough documentation here.
+
+#### Hints
+The rules should be as exclusive as possible to generate as few false positives as possible. I.e. instead of asking for the surface of any way tagged with `highway=*`, the surface should instead only be asked for an inclusive list of roads.
+
+Also, for very detailed information that can be assumed to always have the same ("undefined") answer in many countries, it might be a good idea to limit quests to an inclusive list of certain countries.
+
+### Properties
+
+You can obtain more info about properties by mousing over a property and pressing Ctrl+Q within Android Studio
```

Comment:
I do not know this specific editor but I think the cursor and not the mouse is important for the hotkey.
```suggestion
You can obtain more info about properties by placing the cursor in a property and pressing Ctrl+Q within Android Studio.
```
There are more sentences which misses a dot.

Reply:
Good catch, thanks!